simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/utils/FragmentUtils.kt

```diff
@@ -0,0 +1,23 @@
+package org.nypl.simplified.cardcreator.utils
+
+import androidx.fragment.app.Fragment
+import java.lang.Double
+
+/**
+ * Gets Cache object
+ */
+
+fun Fragment.getCache(): Cache {
+  return Cache(this.requireActivity())
+}
+
+/**
+ * Determines whether or not the user identifier is a barcode or username
+ */
+fun Fragment.isBarcode(identifier: String): Boolean {
+  return try {
+    Double.parseDouble(identifier); true
```

Comment:
You might be able to use `toDoubleOrNull` here:

```kotlin
return identifier.toDoubleOrNull() != null
```

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/to-double-or-null.html